summingbird-online/src/test/scala/com/twitter/summingbird/online/executor/FutureQueueLaws.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ Copyright 2016 Twitter, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package com.twitter.summingbird.online.executor
+
+import com.twitter.conversions.time._
+import com.twitter.summingbird.online.option.{ MaxEmitPerExecute, MaxFutureWaitTime, MaxWaitingFutures }
+import com.twitter.util._
+import org.scalacheck._
+import org.scalacheck.Gen._
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Prop._
+import org.scalatest.WordSpec
+import org.scalatest.concurrent.Eventually
+import org.scalatest.time.{ Seconds, Span }
+import scala.util.{ Failure, Success }
+
+case class NonNegativeShort(get: Short) {
+  require(get >= 0)
+}
+
+class FutureQueueLaws extends Properties("FutureQueue") with Eventually {
+  def genTry[T](implicit arb: Arbitrary[T]): Gen[Try[T]] = Gen.oneOf(arb.arbitrary.map(Return(_)), Arbitrary.arbitrary[java.lang.Exception].map(Throw(_)))
+  implicit def arbTry[T: Arbitrary] = Arbitrary(genTry[T])
+
+  implicit val arbNonNegativeShort: Arbitrary[NonNegativeShort] = Arbitrary(
+    Arbitrary.arbitrary[Short].filter { _ >= 0 }.map { NonNegativeShort }
+  )
+  def twitterToScala[T](t: Try[T]) = t match {
```

Comment:
I've seen this written like 100 times. I finally got it into util, but we might not want to upgrade the dependency just to get it (we are having some GC regressions in some of the latest finagles).